lib/erb_lint/linters/require_script_nonce.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'better_html'
+require 'better_html/tree/tag'
+
+module ERBLint
+  module Linters
+    # Allow inline script tags in ERB that have a nonce attribute.
+    # This only validates inline <script> tags, as well as rails helpers like javascript_tag.
+    class RequireScriptNonce < Linter
+      include LinterRegistry
+
+      def run(processed_source)
+        parser = processed_source.parser
+
+        find_html_script_tags(parser)
+        find_rails_helper_script_tags(parser)
+      end
+
+      private
+
+      def find_html_script_tags(parser)
+        parser.nodes_with_type(:tag).each do |tag_node|
+          tag = BetterHtml::Tree::Tag.from_node(tag_node)
+          next if tag.closing?
+          next unless tag.name == 'script'
+
+          type_attribute = tag.attributes['type']
+          # type attribute = something other than text/javascript
+          next if type_attribute &&
+            type_attribute.value_node.present? &&
+            type_attribute.value_node.to_a[1] != 'text/javascript'
```

Comment:
Should we check `"application/javascript"` as well?

We should also extract this check to a private method with a meaningful name

Reply:
All set! `application/javascript` has been added, and the logic here (and below) has been extracted to private methods.